k8s/install/terraform/rke.tf

```diff
@@ -0,0 +1,547 @@
+resource "random_uuid" "random_cluster_id" {}
+
+resource "aws_instance" "rke_bastion" {
+  ami = var.ami_id
+  instance_type = "t2.micro"
+  key_name = local.key_name
+
+  subnet_id = aws_subnet.public_subnet.id
+  vpc_security_group_ids = [aws_security_group.default.id, aws_security_group.bastion_sg.id]
+  tags = {
+    Name        = "${var.environment}-bastion"
+    Environment = "${var.environment}"
+  }
+}
+resource "aws_eip_association" "rke_bastion_eip" {
+  depends_on = [aws_route_table_association.public, aws_route.public_internet_gateway]
+  network_interface_id = aws_instance.rke_bastion.primary_network_interface_id
+  allocation_id = aws_eip.bastion_eip.id
+}
+
+resource "aws_lb" "rke_master_lb" {
+  depends_on         = [aws_route_table_association.public, aws_route.public_internet_gateway, aws_internet_gateway.ig]
+  name               = "${var.environment}-master-lb"
+  internal           = false
+  load_balancer_type = "network"
+  subnets            = [aws_subnet.public_subnet.id]
+  security_groups    = [aws_security_group.default.id]
+  tags = {
+    Name        = "${var.environment}-master-lb"
+    Environment = "${var.environment}"
+  }
+}
+
+locals {
+  rke_master_lb_hostname = split(".", "${aws_lb.rke_master_lb.dns_name}")[0]
+}
+
+resource "null_resource" "lb_ip_getter" {
+  depends_on = [aws_eip_association.rke_bastion_eip,
+                aws_instance.rke_bastion,
+                aws_lb.rke_master_lb]
+
+  connection {
+    type = "ssh"
+    user = "centos"
+    private_key = file(local.private_key_file)
+    host = aws_eip.bastion_eip.public_ip
+  }
+
+  provisioner "remote-exec" {
+    inline = [
+        "sudo yum install -y bind-utils"
+    ]
+  }
+
+  provisioner "remote-exec" {
+    inline = [
+        "sudo dig ${local.rke_master_lb_hostname} +short > lb_ip.txt"
```

Comment:
This doesn't put the IP in the local file for me,
I manually fixed this by adding the private ip of the LB created every time.